cluster/gce/util.sh

```diff
@@ -765,13 +765,23 @@ function construct-linux-kubelet-flags {
       flags+=" --kubeconfig=/var/lib/kubelet/bootstrap-kubeconfig"
       flags+=" --register-schedulable=false"
     fi
+    if [[ "${MASTER_OS_DISTRIBUTION}" == "ubuntu" ]]; then
+      # Configure the file path for host dns configuration
+      # as ubuntu uses systemd-resolved
+      flags+=" --resolv-conf=/run/systemd/resolve/resolv.conf"
+    fi
   else # For nodes
     flags+=" ${NODE_KUBELET_TEST_ARGS:-}"
     flags+=" --bootstrap-kubeconfig=/var/lib/kubelet/bootstrap-kubeconfig"
     flags+=" --kubeconfig=/var/lib/kubelet/kubeconfig"
     if [[ "${node_type}" == "heapster" ]]; then
         flags+=" ${HEAPSTER_KUBELET_TEST_ARGS:-}"
     fi
+    if [[ "${NODE_OS_DISTRIBUTION}" == "ubuntu" ]]; then
```

Comment:
I guess that this can be added at the beginning of the function and it will apply to all nodes, so we don't have to replicate this code

Reply:
i am expecting more things to pop up (some differences between the master and node), so for better readability, let's leave this as is please.

Reply:
yeah, sure, was more a nit than anything :sweat_smile: